src/tqec/generation/circuit.py

```diff
@@ -70,20 +62,33 @@ def generate_circuit(
     ), "Qubits used in plaquette layers should be instances of cirq.GridQubit."
 
     # Generate the ScheduledCircuit instances for each plaquette instanciation
-    all_scheduled_circuits: list[ScheduledCircuit] = list()
+    all_scheduled_circuits: list[ScheduledCircuit] = []
     plaquette_index: int
-    for plaquette_y, line in enumerate(template_plaquettes):
-        for plaquette_x, plaquette_index in enumerate(line):
-            qubit_x = get_plaquette_starting_index(plaquette_shape.x, plaquette_x)
-            qubit_y = get_plaquette_starting_index(plaquette_shape.y, plaquette_y)
+    for row_index, line in enumerate(template_plaquettes):
+        for column_index, plaquette_index in enumerate(line):
             scheduled_circuit = deepcopy(plaquette_circuits[plaquette_index])
-            qubit_map = {
-                # GridQubit are indexed as (row, col), so (y, x)
-                qubit: qubit + (qubit_y, qubit_x)  # type: ignore
-                for qubit in scheduled_circuit.raw_circuit.all_qubits()
-            }
-            scheduled_circuit.map_to_qubits(qubit_map, inplace=True)
+
+            if plaquette_index > 0:
+                offset = (column_index * increments[0], row_index * increments[1])
+                plaquette = plaquettes[plaquette_index - 1]
+                qubit_map = _create_mapping(plaquette, scheduled_circuit, offset)
+                scheduled_circuit.map_to_qubits(qubit_map, inplace=True)
             all_scheduled_circuits.append(scheduled_circuit)
```

Comment:
In my opinion, the `if` statement should either enclose all the code in the `for`-loop or be removed.

Reply:
Just realized that you were using `plaquettes` and not `plaquette_circuits` here. I read too quickly. The `if` statement was indeed needed, and my comment was wrong.